### ISSUE TITLE
Marking PlutusTx.Builtins.Internal -O is overkill

### ISSUE THREAD

**Issue by treeowl** [input-output-hk/plutus#4194]
Currently,

```haskell
-- This ensures that we don't put *anything* about these functions into the interface
-- file, otherwise GHC can be clever about the ones that are always error, even though
-- they're NOINLINE!
{-# OPTIONS_GHC -O0 #-}
```

Now I don't know what the problem is with GHC being clever like that (that should be documented!), but this is overkill regardless. As far as I can tell, the only thing that's always error is `error :: BuiltinUnit -> a`. Everything else in the module should ideally be analyzed properly. I think the right thing is to just tell GHC exactly what we mean.

```haskell
error :: BuiltinUnit -> a
error (BuiltinUnit ()) = GHC.Exts.lazy (mustBeReplaced "error")
```

Note: in combination with #4193, we'll then want to use `lazy` in the `BuiltinByteString` functions so GHC sees those as strict in their `BuiltinByteString` arguments but not in the underlying `ByteString`s.

**Comment by michaelpj** [comment]
Yes, it's a blunt hammer, but it works. What we really want is https://github.com/ghc-proposals/ghc-proposals/pull/415 (for essentially the same reason as the Clash folks want it).

**Comment by effectfully** [comment]
> Yes, it's a blunt hammer, but it works.

Given this, I'm marking this issue with "Low priority".
